src/app/messagefilter.cpp

```diff
@@ -92,7 +94,15 @@ bool MessageFilter::filterAcceptsRow(int sourceRow, const QModelIndex& sourcePar
         case IrcMessage::Topic:
         case IrcMessage::Numeric:
         case IrcMessage::Names:
-            return m_topicMessages || !dynamic_cast<const MessageModel*>(sourceModel())->buffer()->isChannel();
+            if (m_topicMessages)
+            {
+                if (m_buffer->connection()->connectionCount() == 1)
+                    return false;
```

Comment:
is this wrong way around? this would filter out topic messages when the connection count equals to 1. is IrcConnection::connectionCount() 1 on the very first connection attempt?